pkg/server/container_create.go

```diff
@@ -207,7 +204,7 @@ func (c *criContainerdService) CreateContainer(ctx context.Context, r *runtime.C
 	opts = append(opts,
 		containerd.WithSpec(spec, specOpts...),
 		containerd.WithRuntime(defaultRuntime, nil),
-		containerd.WithContainerLabels(labels))
+		containerd.WithContainerExtension(containerMetadataExtension, &meta))
```

Comment:
We still need to handle versioning ourselves, I think.

Containerd is using json marshal/unmarshal for non-proto message now. Could we update `Encode`/`Decode` in container/sandbox metadata and container status to `MarshalJSON`, and `UnmarshalJSON`? So that golang will call our customized marshal/unmarshal functions which should automatically handle versioning.

Reply:
And please add unit test to make sure `typeurl.Marshal` and `typeurl.Unmarshal` does properly handle the versioning.

Reply:
done